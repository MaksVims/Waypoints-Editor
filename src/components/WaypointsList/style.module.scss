@import "../../styles";

.waypointList {
  position: relative;
  width: 100%;
}

