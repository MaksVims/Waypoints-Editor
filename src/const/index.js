export const GOOGLE_API_KEY = 'AIzaSyBonvH3Y1Y_iIH1fHmvjxsbNH3nCTWQeY8'
export const libraries = ["places"]

export const CONTAINER_STYLE = {
  width: '100%',
  height: '100%'
};

export const mockWaypoint = {
  id: 1,
  geo: {
    address: "lorem",
    place_id: 21
  },
  coords: {
    lat: 12,
    lng: 2
  }
}

export const mockWaypoints = [
  {id: 1, geo: {address: 'lorem', place_id: 1}, coords: {lat: 0, lng: 1}},
  {id: 2, geo: {address: 'lorem1', place_id: 12}, coords: {lat: 3, lng: 4}},
  {id: 3, geo: {address: 'lorem2', place_id: 13}, coords: {lat: 5, lng: 4}},
  {id: 4, geo: {address: 'lorem3', place_id: 14}, coords: {lat: 1, lng: 3}},
]