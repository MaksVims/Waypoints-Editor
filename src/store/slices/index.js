export {default as waypointsReducer} from './waypointsSlice'
export * from './waypointsSlice'