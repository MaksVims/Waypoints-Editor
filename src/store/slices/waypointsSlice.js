import {createSlice} from "@reduxjs/toolkit";

const waypointsSlice = createSlice({
  name: 'waypoints',

  initialState: {
    waypoints: [],
    currentCenter: {lat: 0, lng: 0}
  },

  reducers: {
    addWaypoint: ((state, action) => {
      state.waypoints.push(action.payload)
    }),

    removeWaypoint: (state, action) => {
      state.waypoints = state.waypoints.filter(point => point.id !== action.payload)
    },

    updateWaypoint: (state, action) => {
      state.waypoints = state.waypoints.map(point => {
        if (point.id === action.payload.id) {
          return action.payload
        }
        return point
      })
    },

    reorderedWaypoint: (state, action) => {
      // меняем порядок элементов в массиве
      const {from, to} = action.payload
      const waypoints = state.waypoints.slice()
      const [reorderWaypoint] = waypoints.splice(from, 1)
      waypoints.splice(to, 0, reorderWaypoint)
      state.waypoints = waypoints
    },

    setCenter: (state, action) => {
      state.currentCenter = action.payload
    }
  }
})

export default waypointsSlice.reducer
export const {
  addWaypoint,
  removeWaypoint,
  reorderedWaypoint,
  setCenter,
  updateWaypoint,
} = waypointsSlice.actions