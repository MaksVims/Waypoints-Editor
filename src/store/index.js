import {configureStore} from "@reduxjs/toolkit";
import {waypointsReducer} from "./slices";

export const store = configureStore({
  reducer: {
    'waypoints': waypointsReducer
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat()
})

