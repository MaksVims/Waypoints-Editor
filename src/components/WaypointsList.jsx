import React, {useCallback} from 'react';
import {useDispatch, useSelector} from "react-redux";
import styled from "styled-components";
import {removeWaypoint, reorderedWaypoint} from "../store/slices";
import {DragDropContext, Draggable, Droppable} from "react-beautiful-dnd";

const WaypointsListContainer = styled.div``

const Waypoint = styled.li`
  background-color: #fff;
  cursor: pointer;
  font-size: 21px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 10px;
  border: 1px solid ${props => props.theme.color.light};

  &:not(:last-child) {
    margin-bottom: 10px;
  }

  @media (max-width: ${props => props.theme.media.middle}) {
    font-size: 18px;
  }
`
const WaypointText = styled.p``

const WaypointList = styled.ul`
  position: relative;
  width: 100%;
`

const RemoveButton = styled.button`
  border-radius: 2px;
  font-size: 21px;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({theme}) => theme.color.danger};
  width: 25px;
  height: 25px;
  cursor: pointer;
  color: #fff;
  margin-left: 5px;
  flex-shrink: 0;

  &:hover {
    transform: scale(1.05);
  }

  @media (max-width: ${props => props.theme.media.middle}) {
    width: 22px;
    height: 22px;
  }
`

const WaypointsList = () => {
  const waypoints = useSelector(state => state.waypoints.waypoints)
  const dispatch = useDispatch()

  const handleRemoveWaypoint = useCallback((id) => dispatch(removeWaypoint(id)), [dispatch])

  const handleOnDragEnd = useCallback((result) => {
    // Проверка на drop внутри контейнера
    if (!result.destination) return

    const from = result.source.index
    const to = result.destination.index
    // Изменяем порядок элементов в массиве
    dispatch(reorderedWaypoint({from, to}))
  }, [dispatch])


  return (
    <WaypointsListContainer>
      <DragDropContext onDragEnd={handleOnDragEnd}>
        <Droppable droppableId="waypointList">
          {provided => (
            <WaypointList
              className="waypointList"
              {...provided.droppableProps}
              ref={provided.innerRef}
            >
              {waypoints?.map((point, idx) => (
                <Draggable key={point.id} draggableId={String(point.id)} index={idx}>
                  {provided => (
                    <Waypoint
                      ref={provided.innerRef}
                      {...provided.draggableProps}
                      {...provided.dragHandleProps}
                    >
                      <WaypointText>{point.geo.address}</WaypointText>
                      <RemoveButton onClick={() => handleRemoveWaypoint(point.id)}>-</RemoveButton>
                    </Waypoint>
                  )}
                </Draggable>
              ))}
              {provided.placeholder}
            </WaypointList>
          )}
        </Droppable>
      </DragDropContext>
    </WaypointsListContainer>
  );
};

export default WaypointsList;