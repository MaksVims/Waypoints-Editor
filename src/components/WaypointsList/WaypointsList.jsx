import React, {useCallback} from 'react';
import {useDispatch, useSelector} from "react-redux";
import classes from './style.module.scss'
import {DragDropContext, Draggable, Droppable} from "react-beautiful-dnd";
import {removeWaypoint, reorderedWaypoint} from "../../store/slices";
import WaypointItem from "../WaypointItem/WaypointItem";


const WaypointsList = () => {
  const waypoints = useSelector(state => state?.waypoints?.waypoints)
  const dispatch = useDispatch()

  const handleRemoveWaypoint = useCallback((id) => dispatch(removeWaypoint(id)), [dispatch])

  const handleOnDragEnd = useCallback((result) => {
    // Проверка на drop внутри контейнера
    if (!result.destination) return

    const from = result.source.index
    const to = result.destination.index
    // Изменяем порядок элементов в массиве
    dispatch(reorderedWaypoint({from, to}))
  }, [dispatch])


  return (
    <div>
      <DragDropContext onDragEnd={handleOnDragEnd}>
        <Droppable droppableId={"waypoint-list"}>
          {provided => (
            <ul
              id="waypoint-list"
              className={[classes.waypointList]}
              {...provided.droppableProps}
              ref={provided.innerRef}
            >
              {waypoints?.map((point, idx) => (
                <Draggable key={point.id} draggableId={String(point.id)} index={idx}>
                  {provided => (
                    <li
                      className={classes.dragWrapper}
                      ref={provided.innerRef}
                      {...provided.draggableProps}
                      {...provided.dragHandleProps}
                    >
                      <WaypointItem
                        point={point}
                        handleClick={handleRemoveWaypoint}
                      />
                    </li>
                  )}
                </Draggable>
              ))}
              {provided.placeholder}
            </ul>
          )}
        </Droppable>
      </DragDropContext>
    </div>
  );
};

export default WaypointsList;