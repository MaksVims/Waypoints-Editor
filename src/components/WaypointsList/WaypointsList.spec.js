import {configureStore} from "@reduxjs/toolkit";
import {addWaypoint, waypointsReducer} from "../../store/slices";
import {render, screen} from "@testing-library/react";
import {Provider} from "react-redux";
import WaypointsList from "./WaypointsList";
import {mockWaypoints} from "../../const";

describe(">>> C O M P O N E N T --- WaypointsList", function () {
  let store, waypoints

  beforeEach(() => {
    waypoints = [...mockWaypoints]
    store = configureStore({
      reducer: {'waypoints': waypointsReducer}
    })
  })

  it('+++ should correct render component', function () {
    render(<Provider store={store}><WaypointsList/></Provider>)
    expect(screen.queryByText('lorem')).toBeNull()
    expect(screen.getByRole('list')).toBeInTheDocument()
  });


  it('+++ should update view after added waypoints in redux', function () {
    const {rerender} = render(<Provider store={store}><WaypointsList/></Provider>)
    waypoints.forEach(point => store.dispatch(addWaypoint(point)))

    rerender(<Provider store={store}><WaypointsList/></Provider>)

    expect(screen.getByText('lorem')).toBeInTheDocument()
  });


  it('+++ should equal snapshot', function () {
    const result = render(<Provider store={store}><WaypointsList/></Provider>)
    expect(result).toMatchSnapshot()
  });
})