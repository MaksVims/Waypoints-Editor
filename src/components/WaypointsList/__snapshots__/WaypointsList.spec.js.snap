// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> C O M P O N E N T --- WaypointsList +++ should equal snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div>
        <ul
          class="waypointList"
          data-rbd-droppable-context-id="2"
          data-rbd-droppable-id="waypoint-list"
          id="waypoint-list"
        />
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-2"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-2-hidden-text-8"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="waypointList"
        data-rbd-droppable-context-id="2"
        data-rbd-droppable-id="waypoint-list"
        id="waypoint-list"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`>>> C O M P O N E N T --- WaypointsList +++ should equal snapshot add many items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-2"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div>
        <ul
          class="waypointList"
          data-rbd-droppable-context-id="3"
          data-rbd-droppable-id="waypoint-list"
          id="waypoint-list"
        >
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="1"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="1"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="2"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="2"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem1
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="3"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="3"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem2
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="4"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="4"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem3
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-3"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-3-hidden-text-10"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="waypointList"
        data-rbd-droppable-context-id="3"
        data-rbd-droppable-id="waypoint-list"
        id="waypoint-list"
      >
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="1"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="1"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="2"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="2"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem1
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="3"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="3"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem2
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="4"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="4"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem3
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`>>> C O M P O N E N T --- WaypointsList +++ should equal snapshot after adding many items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-0"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-1"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-2"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div>
      <div>
        <ul
          class="waypointList"
          data-rbd-droppable-context-id="3"
          data-rbd-droppable-id="waypoint-list"
          id="waypoint-list"
        >
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="1"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="1"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="2"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="2"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem1
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="3"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="3"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem2
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
          <li
            aria-describedby="rbd-hidden-text-3-hidden-text-10"
            class="dragWrapper"
            data-rbd-drag-handle-context-id="3"
            data-rbd-drag-handle-draggable-id="4"
            data-rbd-draggable-context-id="3"
            data-rbd-draggable-id="4"
            draggable="false"
            role="button"
            tabindex="0"
          >
            <div
              class="waypoint"
            >
              <p>
                lorem3
              </p>
              <button
                class="removeButton"
              >
                -
              </button>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      id="rbd-announcement-3"
      style="position: absolute; width: 1px; height: 1px; margin: -1px; border: 0px; padding: 0px; overflow: hidden; clip-path: inset(100%);"
    />
    <div
      id="rbd-hidden-text-3-hidden-text-10"
      style="display: none;"
    >
      
  Press space bar to start a drag.
  When dragging you can use the arrow keys to move the item around and escape to cancel.
  Some screen readers may require you to be in focus mode or to use your pass through key

    </div>
  </body>,
  "container": <div>
    <div>
      <ul
        class="waypointList"
        data-rbd-droppable-context-id="3"
        data-rbd-droppable-id="waypoint-list"
        id="waypoint-list"
      >
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="1"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="1"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="2"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="2"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem1
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="3"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="3"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem2
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
        <li
          aria-describedby="rbd-hidden-text-3-hidden-text-10"
          class="dragWrapper"
          data-rbd-drag-handle-context-id="3"
          data-rbd-drag-handle-draggable-id="4"
          data-rbd-draggable-context-id="3"
          data-rbd-draggable-id="4"
          draggable="false"
          role="button"
          tabindex="0"
        >
          <div
            class="waypoint"
          >
            <p>
              lorem3
            </p>
            <button
              class="removeButton"
            >
              -
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
